metadata-service/factories/src/main/java/com/linkedin/gms/factory/telemetry/Telemetry.java

```diff
@@ -0,0 +1,142 @@
+package com.linkedin.gms.factory.telemetry;
+
+import com.mixpanel.mixpanelapi.MessageBuilder;
+import com.mixpanel.mixpanelapi.MixpanelAPI;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.util.UUID;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static java.lang.Boolean.parseBoolean;
+
+@Slf4j
+final class Telemetry {
+    static final String MIXPANEL_TOKEN = "5ee83d940754d63cacbf7d34daa6f44a";
+    static final String DATAHUB_FOLDER_PATH = System.getProperty("user.home") + "/.datahub";
+    static final String CONFIG_FILE_PATH = DATAHUB_FOLDER_PATH + "/telemetry-config.json";
+
+    static boolean enabled = true;
+    static String clientId;
+
+    private static MixpanelAPI mixpanel;
+    private static MessageBuilder mixpanelBuilder;
+
+    static {
+
+        try {
+            File configFile = new File(CONFIG_FILE_PATH);
+
+            if (!configFile.exists()) {
+                clientId = UUID.randomUUID().toString();
+                updateConfig();
+            } else {
+                loadConfig();
+            }
+
+            if (enabled) {
+
+                // initialize MixPanel instance and message builder
+                mixpanel = new MixpanelAPI();
+                mixpanelBuilder = new MessageBuilder(MIXPANEL_TOKEN);
+
+                // set user-level properties
+                JSONObject props = new JSONObject();
+                props.put("java_version", System.getProperty("java.version"));
+                props.put("os", System.getProperty("os.name"));
+                JSONObject update = mixpanelBuilder.set(clientId, props);
+                mixpanel.sendMessage(update);
+            }
+
+        } catch (Exception e) {
```

Comment:
catching whole chunk of code is not recommended. What are we catching here? If it's just whether the file exists on the path, we should try catch just that